src/_pytest/main.py

```diff
@@ -991,11 +991,19 @@ def search_pypath(module_name: str) -> str | None:
     # ValueError: not a module name
     except (AttributeError, ImportError, ValueError):
         return None
-    if spec is None or spec.origin is None or spec.origin == "namespace":
+
+    if spec is None:
         return None
-    elif spec.submodule_search_locations:
-        return os.path.dirname(spec.origin)
+    elif (
+        spec.submodule_search_locations is not None
+        and len(spec.submodule_search_locations) > 0
+    ):
+        # If submodule_search_locations is set, it's a package (regular or namespace).
+        # Typically there is a single entry, but documentation claims it can be empty too
+        #  (e.g. if the package has no physical location).
+        return spec.submodule_search_locations[0]
```

Comment:
Perhaps instead of just returning it blindly, we should check if this is a directory before returning?

Reply:
So I looked at it:
- in my experience of working with namespace packages, this is always a directory. I guess it's possible for it not to be if some sort of obscure package loader is used, but I'm not sure pytest would work with such packages anyway?
- either way even if it's not there isn't much we can do.  
    If `spec.submodule_search_locations` is set at all, then `spec.origin` will be `None`, so we can't even fall back onto `os.path.dirname(spec.origin)`.

IMO it's fine to keep it as is, especially considering now this is hidden behind `consider_namespace_packages`